units/host_execute.go

```diff
@@ -89,10 +89,11 @@ func (j *hostExecuteJob) Run(ctx context.Context) {
 				args = append(args, fmt.Sprintf("--user=%s", j.SudoUser))
 			}
 		}
-		args = append(args, j.host.Distro.ShellBinary(), "-l", "-c", j.Script)
+		args = append(args, j.host.Distro.ShellBinary(), "-s")
```

Comment:
I think we still need `-l` here. `RunSSHShellScript` is written for provisioning and the provisioning SSH commands are written in such a way to not depend on the particular shell context (except where it's necessary to do so). I imagine some scripts may already assume that they're running in a login session (i.e. a shell environment similar to what SSH will give them). However, the way this is implemented, it is running the command via exec (i.e. Jasper), so it will not have any login session unless it's explicitly set.

Reply:
I imagine the `-s` is a way to overcome [this issue on Windows](https://github.com/dotnet/msbuild/issues/2530). Can we also add a comment that `-s` is necessary due to exec string length limitations?

Reply:
done and done

Reply:
I think that was Max's conclusion too.